### PR TITLE
fix click on layer list

### LINKED ISSUE #304: error when clicking in layers list
## 🐛 Bug

program crashes when i click in the layers list

## To Reproduce

1. open any example
2. click in the layers list

```python
WARNING: Traceback (most recent call last):
  File "/Users/kiraevans/Projects/napari/napari/components/_layers_list/view/main.py", line 115, in mouseMoveEvent
    distance = (event.pos() - self.drag_start_position).manhattanLength()
TypeError: unsupported operand type(s) for -: 'QPoint' and 'tuple'
```

## Expected behavior

it doesnt crash

## Environment

 - napari Version (e.g., 1.0): master
 - OS (e.g., Linux): OSX
 - Python version: 3.7
 - Any other relevant information:

`conda list | grep qt` output:
```
pyqt                      5.9.2            py37h655552a_2
pyqt5                     5.12.1                   pypi_0    pypi
pyqt5-sip                 4.19.15                  pypi_0    pypi
qt                        5.9.7                h468cd18_1
qtawesome                 0.5.7                    pypi_0    pypi
qtconsole                 4.4.3                    pypi_0    pypi
qtpy                      1.7.0                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.2                    pypi_0    pypi
```

### PR DESCRIPTION
**Pull request by sofroniewn**
# Description
This PR fixes #304 (hopefully). It's a bit hard for me to know as I don't have this problem on my machine, but I think this may do it. Maybe @kne42 can test

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [x] run `example/add_image.py` and click on layer list

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by kne42** [comment]
new error:
```python
WARNING: Traceback (most recent call last):
  File "/Users/kiraevans/Projects/napari/napari/components/_layers_list/view/main.py", line 151, in dragMoveEvent
    index = self.layers.index(layer_name)
  File "/Users/kiraevans/Projects/napari/napari/util/list/_typed.py", line 76, in index
    raise KeyError(f'could not find element {q} was referencing')
KeyError: 'could not find element  was referencing'
```

**Comment by sofroniewn** [comment]
@kne42 this should be fixed now, can you check again, thanks!

**Comment by kne42** [comment]
still broken

**Comment by kne42** [comment]
ok so i investigated and the error is different now – due to a bug in the implementation of custom lists

however, if you look more closely at the error message, you'll notice that the layer name is nothing, just an empty string

this is because your fix does not address the root of the problem, which is that this code is getting called at all despite the fact that I am not clicking on any layers. i am clicking in an area inside the layer list but it is empty space, so the related code should not be run at all

**Comment by sofroniewn** [comment]
I was finally able to reproduce this bug, it required initiating a drag outside the layer properties not just clicking. I have now fixed the root cause by preventing the drag from initiating in this situation. This is ready for re-review from @kne42

File: napari/components/_layers_list/view/main.py [hunk 1]
```diff
@@ -148,16 +152,17 @@ def dragMoveEvent(self, event):
...
-        # If the widget being dragged hasn't moved above or below any other
-        # widgets then don't highlight any dividers
-        selected = (not (insert == index) and not (insert-1 == index))
-        # Set the selected state of all the dividers
-        for i in range(0, self.vbox_layout.count(), 2):
-            if i == 2*divider_index:
-                self.vbox_layout.itemAt(i).widget().setSelected(selected)
-            else:
-                self.vbox_layout.itemAt(i).widget().setSelected(False)
+        if layer_name in self.layers:
```
Comment (kne42) [inline napari/components/_layers_list/view/main.py]: this should be removed – we should always check if what's being dragged is actually a layer, in which case, its name should _always_ be in the layers list anyways

**Review by kne42 — APPROVED** [review]
looks great! :)

File: napari/components/_layers_list/view/main.py (hunk 1 above)
Comment (sofroniewn) [reply napari/components/_layers_list/view/main.py]: done

**Pull request merged**